src/core/Object3D.js

```diff
@@ -94,6 +94,7 @@ class Object3D extends EventDispatcher {
 
 		this.matrix = new Matrix4();
 		this.matrixWorld = new Matrix4();
+		this.matrixWorldPrevious = new Matrix4();
```

Comment:
I wonder if we could avoid adding `matrixWorldPrevious` to `Object3D` and hide it inside the renderer. Maybe below the following line: https://github.com/mrdoob/three.js/blob/1e0cdd9c6304dac238e9d6891652c1ce81f38580/src/renderers/WebGLRenderer.js#L923
something like:
```js
scene.traverse( function( object  ) {

    if ( object.isMesh && object.material.isMeshVelocityMaterial ) {

        const objectProperties = properties.get( object );
        
        if ( objectProperties.matrixWorldPrevious === undefined ) objectProperties.matrixWorldPrevious = new Matrix4();
        
        objectProperties.matrixWorldPrevious.copy( object.matrixWorld );

    }

) };

Reply:
> Setting `material.needsUpdate = true;` is not necessary when changing `material.side`.
> 
> However, toggling between `BasicDepthPacking` and `RGBADepthPacking` does not update the material.
> 
> EDiT: It appears the renderer treats `parameters.depthPacking` as a boolean, and it is not.

We should not be using depthPacking on this material anyhow.